plugins/modules/aci_node_block.py

```diff
@@ -0,0 +1,390 @@
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(
+        switch_profile=dict(
+            type="str",
+            aliases=[
+                "leaf_profile_name",
+                "leaf_profile",
+                "switch_profile_name",
+                "spine_switch_profile",
+                "spine_switch_profile_name",
+            ],
+        ),  # Not required for querying all objects
+        access_port_selector=dict(
+            type="str",
+            aliases=[
+                "access_port_selector_name",
+                "port_selector",
+                "port_selector_name",
+            ],
+        ),  # Not required for querying all objects
+        node_block=dict(type="str", aliases=["node_block_name", "name"]),  # Not required for querying all objects
+        description=dict(type="str", aliases=["node_block_description"]),
+        from_port=dict(type="str", aliases=["from", "from_port_range"]),
+        to_port=dict(type="str", aliases=["to", "to_port_range"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        type_node=dict(type="str", default="leaf", choices=["leaf", "spine"], aliases=["type"]),  # This parameter is not required for querying all objects
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["switch_profile", "access_port_selector", "node_block"]],
+            ["state", "present", ["switch_profile", "access_port_selector", "node_block"]],
+        ],
+    )
+
+    switch_profile = module.params.get("switch_profile")
+    access_port_selector = module.params.get("access_port_selector")
+    node_block = module.params.get("node_block")
+    description = module.params.get("description")
+    from_port = module.params.get("from_port")
+    to_port = module.params.get("to_port")
+    state = module.params.get("state")
+    type_node = module.params.get("type")
+
+    aci = ACIModule(module)
+
+    subclass_1 = dict(
+        aci_class="infraNodeP",
+        aci_rn="nprof-{0}".format(switch_profile),
+        module_object=switch_profile,
+        target_filter={"name": switch_profile},
+    )
+    subclass_2 = dict(
+        aci_class="infraLeafS",
+        aci_rn="leaves-{0}-typ-range".format(access_port_selector),
+        module_object=access_port_selector,
+        target_filter={"name": access_port_selector},
+    )
+    if type_node == "spine":
```

Comment:
should we move the above subclass assignments into a if else condition?

Reply:
yes, I think you're right